src/lib/storage/matches_all_pos_list.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "./chunk.hpp"
+#include "abstract_pos_list.hpp"
+
+namespace opossum {
+
+class MatchesAllPosList : public AbstractPosList {
+ public:
+  explicit MatchesAllPosList(std::shared_ptr<const Chunk> common_chunk, const ChunkID common_chunk_id)
+      : _common_chunk(common_chunk),
+        _common_chunk_id(common_chunk_id),
+        _common_chunk_size_on_creation(common_chunk->size()) {}
+
+  MatchesAllPosList& operator=(MatchesAllPosList&& other) = default;
+
+  MatchesAllPosList() = delete;
+
+  bool references_single_chunk() const final { return true; }
+
+  ChunkID common_chunk_id() const final {
+    DebugAssert(_common_chunk_id != INVALID_CHUNK_ID, "common_chunk_id called on invalid chunk id");
+    return _common_chunk_id;
+  }
+
+  RowID operator[](size_t n) const final {
+    DebugAssert(_common_chunk_id != INVALID_CHUNK_ID, "operator[] called on invalid chunk id");
+    return RowID{_common_chunk_id, static_cast<ChunkOffset>(n)};
+  }
+
+  bool empty() const final { return size() == 0; }
+
+  size_t size() const final { return _common_chunk_size_on_creation; }
+
+  size_t memory_usage(const MemoryUsageCalculationMode) const final { return sizeof *this; }
+
+  PosListIterator<const MatchesAllPosList*, RowID> begin() const {
```

Comment:
In theory, a random access iterator (which we were aiming for here) should be default constructible. Currently, in our code, this never happens (default ctor is deleted) -> Currently, it could be a reference.

So: should we make it a reference?

Reply:
Hmm, I just remembered iterators being default-constructible was one of the requirements for the LegacyRandomAccessIterator (inherited from LegacyForwardIterator). If you already deleted the default constructor, changing this to a reference should not make a difference, so yes.

Reply:
We cannot store a reference to the PosList because iterators need to be copy assignable. We have to use a pointer.